src/components/Tool/FixCommitmentInfo/FixCommitmentInfo.tsx

```diff
@@ -65,122 +56,185 @@ const useStyles = makeStyles()((theme: Theme) => ({
-    useContactFiltersQuery({
-      variables: {
-        accountListId,
-      },
-      context: {
-        doNotBatch: true,
-      },
-    });
+
   const [updateInvalidStatus, { loading: updating }] =
-    useUpdateInvalidStatusMutation();
-
-  const contactStatuses = contactFilterGroups?.accountList?.contactFilterGroups
-    ? (
-        contactFilterGroups.accountList.contactFilterGroups
-          .find((group) => group?.filters[0]?.filterKey === 'status')
-          ?.filters.find(
-            (filter: { filterKey: string }) => filter.filterKey === 'status',
-          ) as MultiselectFilter
-      ).options?.filter(
-        (status) =>
-          status.value !== 'NULL' &&
-          status.value !== 'HIDDEN' &&
-          status.value !== 'ACTIVE',
-      )
-    : [{ name: '', value: '' }];
-  //TODO: Make currency field a select element
-
-  const updateContact = async (
-    id: string,
-    change: boolean,
-    status?: string,
-    pledgeCurrency?: string,
-    pledgeAmount?: number,
-    pledgeFrequency?: string,
-  ): Promise<void> => {
-    const attributes = change
-      ? {
-          id,
-          status: status as StatusEnum,
-          pledgeAmount,
-          pledgeCurrency,
-          pledgeFrequency: pledgeFrequency as PledgeFrequencyEnum,
+    useUpdateStatusMutation();
+
+  const contactStatuses = useMemo(() => {
+    return Object.values(StatusEnum).map((value) =>
+      getLocalizedContactStatus(t, value),
+    );
+  }, [t]);
+
+  const formatSuggestedChanges = (
+    suggestedChanges: SuggestedChangesType | string | undefined | null,
+  ) => {
+    if (typeof suggestedChanges !== 'string') {
+      return {};
+    }
+
+    try {
+      return JSON.parse(suggestedChanges);
+    } catch (error) {
+      return {};
+    }
+  };
+
+  const updateContact = async (): Promise<void> => {
+    let attributes;
+
+    switch (modalState.updateType) {
+      case 'CHANGE':
+        attributes = {
+          id: modalState.contact.id,
+          status: modalState.contact.status,
+          pledgeAmount: modalState.contact.pledgeAmount,
+          pledgeCurrency: modalState.contact.pledgeCurrency,
+          pledgeFrequency: modalState.contact.pledgeFrequency,
+          statusValid: true,
+        };
+        break;
+      case 'DONT_CHANGE':
+        attributes = {
+          id: modalState.contact.id,
           statusValid: true,
-        }
-      : { id, statusValid: true };
+        };
+        break;
+      case 'HIDE':
+        attributes = {
+          id: modalState.contact.id,
+          status: 'NEVER_ASK' as StatusEnum,
+        };
+        break;
+    }
+
     await updateInvalidStatus({
       variables: {
         accountListId,
         attributes,
       },
+      update: (cache) => {
+        cache.evict({ id: `Contact:${modalState.contact.id}` });
+      },
+      onError() {
+        enqueueSnackbar(
+          t(`Error updating ${modalState.contact.name}'s commitment info`),
```

Comment:
You need to do interpolation like this. Same with the success message.

```suggestion
          t('Error updating {{name}}'s commitment info', {name: modalState.contact.name}),
```